plugins/lsp/lspsaga.nix

```diff
@@ -7,220 +7,651 @@
+          width = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.6" ''
+            Defines float window width.
+          '';
+
+          height = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.5" ''
+            Defines float window height.
+          '';
+
+          keys = {
+            edit = mkKeymapOption "<C-c>o" "edit";
+
+            vsplit = mkKeymapOption "<C-c>v" "vsplit";
+
+            split = mkKeymapOption "<C-c>i" "split";
+
+            tabe = mkKeymapOption "<C-c>t" "tabe";
+
+            quit = mkKeymapOption "q" "quit";
+
+            close = mkKeymapOption "<C-c>k" "close";
+          };
         };
 
-        findReference = mkOption {
-          type = types.nullOr types.str;
-          default = null;
-          description = "Find reference icon";
+        rename = {
+          inSelect = helpers.defaultNullOpts.mkBool true ''
+            Default is `true`.
+            Whether the name is selected when the float opens.
+
+            In some situation, just like want to change one or less characters, `inSelect` is not so
+            useful.
+            You can tell the Lspsaga to start in normal mode using an extra argument like
+            `:Lspsaga lsp_rename mode=n`.
+          '';
+
+          autoSave = helpers.defaultNullOpts.mkBool false ''
+            Auto save file when the rename is done.
+          '';
+
+          projectMaxWidth =
+            helpers.defaultNullOpts.mkNullable
+            (types.numbers.between 0.0 1.0)
+            "0.5"
+            "Width for the `project_replace` float window.";
+
+          projectMaxHeight =
+            helpers.defaultNullOpts.mkNullable
+            (types.numbers.between 0.0 1.0)
+            "0.5"
+            "Height for the `project_replace` float window.";
+
+          keys = {
+            quit = mkKeymapOption "<C-k>" "Quit rename window or `project_replace` window.";
+
+            exec = mkKeymapOption "<CR>" ''
+              Execute rename in `rename` window or execute replace in `project_replace` window.
+            '';
+
+            select = mkKeymapOption "x" ''
+              Select or cancel select item in `project_replace` float window.
+            '';
+          };
         };
 
-        definitionPreview = mkOption {
-          type = types.nullOr types.str;
-          default = null;
-          description = "Definition preview icon";
+        symbolInWinbar = {
+          enable = helpers.defaultNullOpts.mkBool true "Enable.";
+
+          separator = helpers.defaultNullOpts.mkStr " › " "Separator character.";
+
+          hideKeyword = helpers.defaultNullOpts.mkBool false "Hide keyword.";
+
+          showFile = helpers.defaultNullOpts.mkBool true "Show file.";
+
+          folderLevel = helpers.defaultNullOpts.mkInt 1 "Folder level.";
+
+          colorMode = helpers.defaultNullOpts.mkBool true "Color mode.";
+
+          dely = helpers.defaultNullOpts.mkInt 300 "Delay.";
```

Comment:
typo ? (delay)

Reply:
Yes, but the option is mistyped upstream.
I will correct it in the module though.